src/NumericalAlgorithms/SphericalHarmonics/YlmSpherepack.hpp

```diff
@@ -437,13 +441,14 @@ class YlmSpherepack {
                                 size_t physical_stride = 1,
                                 size_t physical_offset = 0,
                                 bool loop_over_offset = false) const;
-  void calculate_interpolation_data() const;
-  void fill_scalar_work_arrays() const;
-  void fill_vector_work_arrays() const;
+  void calculate_collocation_points();
+  void calculate_interpolation_data();
+  void fill_scalar_work_arrays();
+  void fill_vector_work_arrays();
   size_t l_max_, m_max_, n_theta_, n_phi_;
   size_t spectral_size_;
   mutable YlmSpherepack_detail::MemoryPool memory_pool_;
-  mutable YlmSpherepack_detail::Storage storage_;
+  YlmSpherepack_detail::ConstStorage storage_;
```

Comment:
I got confused that this is named `ConstStorage` but is not actually const, and many member functions retrieve non-const references into it. Can you make this member variable `const`? Maybe move the functions that fill it (such as `calculate_collocation_points`) to `YlmSpherepack_detail` and call them in the constructor of `ConstStorage`?

Reply:
Can I instead rename `ConstStorage` to `Storage`?

I tried making the member variable `const`, and I am going down a rabbit hole of defining custom copy-assignment operators (and copy constructors, move constructors, etc) for many classes.  Apparently `const` member functions don't play well with copy assignment.  And then I saw this: https://stackoverflow.com/questions/4136156

The only member functions that should get non-const refs to `storage_` are the non-const member functions.

Reply:
In general, I've found it's a huge pain to have `const` member variables, as you stated. I don't see an issue with the current design, to be honest. I'd be fine with naming it to `const_storage_` as the member variable too... I don't think this is generally worth the effort. The design is correct, the comments I think are clear

Reply:
Ok, perhaps make the refs into storage_ const in the member functions, meaning `auto& cos_theta = ...` -> `const auto& cos_theta = ...`?

Reply:
Seems reasonable, but if the member functions are `const` (I haven't check if they all are) then `const auto&` and `auto&` would be equivalent (except for readability)